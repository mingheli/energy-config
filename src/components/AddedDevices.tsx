import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faMinusCircle, faArrowUp, faArrowDown } from "@fortawesome/free-solid-svg-icons";
import { AddedDevicesProps, Device } from "../types/commonTypes";
import { formatPrice } from '../utils/commonUtils';
import "../css/ConfigTable.scss";

const AddedDevices: React.FC<AddedDevicesProps> = ({ addedDevices, onMinus, onMoveUp, onMoveDown, onSelectDevice, currentDevice }) => {
    if (!addedDevices) {
        return <div>No devices available</div>;
    }
    return (

        <div className="table-choose-devices">
            <div className="header">Added Devices</div>
            <table>
                <tr>
                    <th className="header-cell">Device Name</th>
                    <th className="header-cell">Floor Dimension</th>
                    <th className="header-cell">Energy</th>
                    <th className="header-cell">Cost</th>
                    <th className="header-cell">Release Year</th>
                    <th className="header-cell"><FontAwesomeIcon icon={faMinusCircle} /></th>
                    <th className="header-cell"><FontAwesomeIcon icon={faArrowUp} /></th>
                    <th className="header-cell"><FontAwesomeIcon icon={faArrowDown} /></th>
                </tr>

                {addedDevices.map((device, index) => (
                    <tr className={`${currentDevice?.id === device.id ? "highlight" : ""}`} key={index} onClick={() => onSelectDevice(device)}>
                        <td className="data-cell">{device.name}</td>
                        <td className="data-cell">{device.floorDimension}</td>
                        <td className="data-cell">{`${device.energy} MWh`}</td>
                        <td className="data-cell">{formatPrice((device.cost))}</td>
                        <td className="data-cell">{device.releaseDate}</td>
                        <td className="data-cell" onClick={() => onMinus(device, index)}><FontAwesomeIcon icon={faMinusCircle} className="minus" /></td>
                        <td className="data-cell" onClick={() => onMoveUp(device, index)}>{index > 0 && <FontAwesomeIcon icon={faArrowUp} className="moveup" />}</td>
                        <td className="data-cell" onClick={() => onMoveDown(device, index)}>{index < addedDevices.length - 1 && <FontAwesomeIcon icon={faArrowDown} className="movedown" />}</td>
                    </tr>
                ))}

            </table>
        </div>


    )
}
export default AddedDevices;