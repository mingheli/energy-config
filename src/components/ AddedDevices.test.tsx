import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import AddedDevices from './AddedDevices';

const mockDevices = [
    { name: 'Device 1', floorDimension: '10x10', energy: 100, cost: '1000', releaseDate: '2020' },
    { name: 'Device 2', floorDimension: '20x20', energy: 200, cost: '2000', releaseDate: '2021' }
];

const mockFormatPrice = (price) => `$${price}`;
const mockOnMinus = jest.fn();
const mockOnMoveUp = jest.fn();
const mockOnMoveDown = jest.fn();
const mockOnSelectDevice = jest.fn();

describe('AddedDevices Component', () => {
    test('renders device information correctly', () => {
        const { container } = render(
            <AddedDevices
                addedDevices={mockDevices}
                formatPrice={mockFormatPrice}
                onMinus={mockOnMinus}
                onMoveUp={mockOnMoveUp}
                onMoveDown={mockOnMoveDown}
                onSelectDevice={mockOnSelectDevice}
            />
        );

        expect(screen.getByText('Device 1')).toBeInTheDocument();
        expect(screen.getByText('10x10')).toBeInTheDocument();
        expect(screen.getByText('100 MWh')).toBeInTheDocument();
        expect(screen.getByText('2020')).toBeInTheDocument();

        expect(screen.getByText('Device 2')).toBeInTheDocument();
        expect(screen.getByText('20x20')).toBeInTheDocument();
        expect(screen.getByText('200 MWh')).toBeInTheDocument();
        expect(screen.getByText('2021')).toBeInTheDocument();

        // Generate snapshot
        expect(container).toMatchSnapshot();
    });

    test('calls onMinus when minus icon is clicked', () => {
        const { container } = render(
            <AddedDevices
                addedDevices={mockDevices}
                formatPrice={mockFormatPrice}
                onMinus={mockOnMinus}
                onMoveUp={mockOnMoveUp}
                onMoveDown={mockOnMoveDown}
                onSelectDevice={mockOnSelectDevice}
            />
        );

        fireEvent.click(container.querySelector('.minus'));

        // Generate snapshot
        expect(container).toMatchSnapshot();
    });

    test('calls onMoveUp when up arrow icon is clicked', () => {
        const { container } = render(
            <AddedDevices
                addedDevices={mockDevices}
                formatPrice={mockFormatPrice}
                onMinus={mockOnMinus}
                onMoveUp={mockOnMoveUp}
                onMoveDown={mockOnMoveDown}
                onSelectDevice={mockOnSelectDevice}
            />
        );

        fireEvent.click(container.querySelector('.moveup'));
        expect(mockOnMoveUp).toHaveBeenCalledWith(mockDevices[1], 1);

        // Generate snapshot
        expect(container).toMatchSnapshot();
    });

    test('calls onMoveDown when down arrow icon is clicked', () => {
        const { container } = render(
            <AddedDevices
                addedDevices={mockDevices}
                formatPrice={mockFormatPrice}
                onMinus={mockOnMinus}
                onMoveUp={mockOnMoveUp}
                onMoveDown={mockOnMoveDown}
                onSelectDevice={mockOnSelectDevice}
            />
        );

        fireEvent.click(container.querySelector('.movedown'));
        expect(mockOnMoveDown).toHaveBeenCalledWith(mockDevices[0], 0);

        // Generate snapshot
        expect(container).toMatchSnapshot();
    });
});