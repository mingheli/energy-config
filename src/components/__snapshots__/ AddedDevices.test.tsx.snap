// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddedDevices Component calls onMinus when minus icon is clicked 1`] = `
<div>
  <div
    class="table-choose-devices"
  >
    <div
      class="header"
    >
      Added Devices
    </div>
    <table>
      <tr>
        <th
          class="header-cell"
        >
          Device Name
        </th>
        <th
          class="header-cell"
        >
          Floor Dimension
        </th>
        <th
          class="header-cell"
        >
          Energy
        </th>
        <th
          class="header-cell"
        >
          Cost
        </th>
        <th
          class="header-cell"
        >
          Release Year
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus "
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up "
            data-icon="arrow-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2 160 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-306.7L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
              fill="currentColor"
            />
          </svg>
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-down "
            data-icon="arrow-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </th>
      </tr>
      <tr
        class="highlight"
      >
        <td
          class="data-cell"
        >
          Device 1
        </td>
        <td
          class="data-cell"
        >
          10x10
        </td>
        <td
          class="data-cell"
        >
          100 MWh
        </td>
        <td
          class="data-cell"
        >
          $1,000.00
        </td>
        <td
          class="data-cell"
        >
          2020
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus minus"
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        />
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-down movedown"
            data-icon="arrow-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </td>
      </tr>
      <tr
        class="highlight"
      >
        <td
          class="data-cell"
        >
          Device 2
        </td>
        <td
          class="data-cell"
        >
          20x20
        </td>
        <td
          class="data-cell"
        >
          200 MWh
        </td>
        <td
          class="data-cell"
        >
          $2,000.00
        </td>
        <td
          class="data-cell"
        >
          2021
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus minus"
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up moveup"
            data-icon="arrow-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2 160 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-306.7L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        />
      </tr>
    </table>
  </div>
</div>
`;

exports[`AddedDevices Component calls onMoveDown when down arrow icon is clicked 1`] = `
<div>
  <div
    class="table-choose-devices"
  >
    <div
      class="header"
    >
      Added Devices
    </div>
    <table>
      <tr>
        <th
          class="header-cell"
        >
          Device Name
        </th>
        <th
          class="header-cell"
        >
          Floor Dimension
        </th>
        <th
          class="header-cell"
        >
          Energy
        </th>
        <th
          class="header-cell"
        >
          Cost
        </th>
        <th
          class="header-cell"
        >
          Release Year
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus "
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up "
            data-icon="arrow-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2 160 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-306.7L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
              fill="currentColor"
            />
          </svg>
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-down "
            data-icon="arrow-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </th>
      </tr>
      <tr
        class="highlight"
      >
        <td
          class="data-cell"
        >
          Device 1
        </td>
        <td
          class="data-cell"
        >
          10x10
        </td>
        <td
          class="data-cell"
        >
          100 MWh
        </td>
        <td
          class="data-cell"
        >
          $1,000.00
        </td>
        <td
          class="data-cell"
        >
          2020
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus minus"
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        />
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-down movedown"
            data-icon="arrow-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </td>
      </tr>
      <tr
        class="highlight"
      >
        <td
          class="data-cell"
        >
          Device 2
        </td>
        <td
          class="data-cell"
        >
          20x20
        </td>
        <td
          class="data-cell"
        >
          200 MWh
        </td>
        <td
          class="data-cell"
        >
          $2,000.00
        </td>
        <td
          class="data-cell"
        >
          2021
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus minus"
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up moveup"
            data-icon="arrow-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2 160 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-306.7L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        />
      </tr>
    </table>
  </div>
</div>
`;

exports[`AddedDevices Component calls onMoveUp when up arrow icon is clicked 1`] = `
<div>
  <div
    class="table-choose-devices"
  >
    <div
      class="header"
    >
      Added Devices
    </div>
    <table>
      <tr>
        <th
          class="header-cell"
        >
          Device Name
        </th>
        <th
          class="header-cell"
        >
          Floor Dimension
        </th>
        <th
          class="header-cell"
        >
          Energy
        </th>
        <th
          class="header-cell"
        >
          Cost
        </th>
        <th
          class="header-cell"
        >
          Release Year
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus "
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up "
            data-icon="arrow-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2 160 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-306.7L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
              fill="currentColor"
            />
          </svg>
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-down "
            data-icon="arrow-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </th>
      </tr>
      <tr
        class="highlight"
      >
        <td
          class="data-cell"
        >
          Device 1
        </td>
        <td
          class="data-cell"
        >
          10x10
        </td>
        <td
          class="data-cell"
        >
          100 MWh
        </td>
        <td
          class="data-cell"
        >
          $1,000.00
        </td>
        <td
          class="data-cell"
        >
          2020
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus minus"
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        />
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-down movedown"
            data-icon="arrow-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </td>
      </tr>
      <tr
        class="highlight"
      >
        <td
          class="data-cell"
        >
          Device 2
        </td>
        <td
          class="data-cell"
        >
          20x20
        </td>
        <td
          class="data-cell"
        >
          200 MWh
        </td>
        <td
          class="data-cell"
        >
          $2,000.00
        </td>
        <td
          class="data-cell"
        >
          2021
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus minus"
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up moveup"
            data-icon="arrow-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2 160 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-306.7L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        />
      </tr>
    </table>
  </div>
</div>
`;

exports[`AddedDevices Component renders device information correctly 1`] = `
<div>
  <div
    class="table-choose-devices"
  >
    <div
      class="header"
    >
      Added Devices
    </div>
    <table>
      <tr>
        <th
          class="header-cell"
        >
          Device Name
        </th>
        <th
          class="header-cell"
        >
          Floor Dimension
        </th>
        <th
          class="header-cell"
        >
          Energy
        </th>
        <th
          class="header-cell"
        >
          Cost
        </th>
        <th
          class="header-cell"
        >
          Release Year
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus "
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up "
            data-icon="arrow-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2 160 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-306.7L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
              fill="currentColor"
            />
          </svg>
        </th>
        <th
          class="header-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-down "
            data-icon="arrow-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </th>
      </tr>
      <tr
        class="highlight"
      >
        <td
          class="data-cell"
        >
          Device 1
        </td>
        <td
          class="data-cell"
        >
          10x10
        </td>
        <td
          class="data-cell"
        >
          100 MWh
        </td>
        <td
          class="data-cell"
        >
          $1,000.00
        </td>
        <td
          class="data-cell"
        >
          2020
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus minus"
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        />
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-down movedown"
            data-icon="arrow-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 470.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 370.8 224 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 306.7L54.6 265.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </td>
      </tr>
      <tr
        class="highlight"
      >
        <td
          class="data-cell"
        >
          Device 2
        </td>
        <td
          class="data-cell"
        >
          20x20
        </td>
        <td
          class="data-cell"
        >
          200 MWh
        </td>
        <td
          class="data-cell"
        >
          $2,000.00
        </td>
        <td
          class="data-cell"
        >
          2021
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-minus minus"
            data-icon="circle-minus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM184 232l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up moveup"
            data-icon="arrow-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2 160 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-306.7L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
              fill="currentColor"
            />
          </svg>
        </td>
        <td
          class="data-cell"
        />
      </tr>
    </table>
  </div>
</div>
`;
